client/components/payment-activity/date-range-picker.tsx

```diff
@@ -0,0 +1,155 @@
+		displayKey: __( 'Last 7 days', 'woocommerce-payments' ),
+	},
+	last_4_weeks: {
+		start: now
+			.clone()
+			.subtract( 4, 'weeks' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 4 weeks', 'woocommerce-payments' ),
+	},
+	last_3_months: {
+		start: now
+			.clone()
+			.subtract( 3, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 3 months', 'woocommerce-payments' ),
+	},
+	last_12_months: {
+		start: now
+			.clone()
+			.subtract( 12, 'months' )
+			.set( { hour: 0, minute: 0, second: 0, millisecond: 0 } ),
+		end: yesterdayEndOfDay,
+		displayKey: __( 'Last 12 months', 'woocommerce-payments' ),
+	},
+	month_to_date: {
+		start: now.clone().startOf( 'month' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Month to date', 'woocommerce-payments' ),
+	},
+	quarter_to_date: {
+		start: now.clone().startOf( 'quarter' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Quarter to date', 'woocommerce-payments' ),
+	},
+	year_to_date: {
+		start: now.clone().startOf( 'year' ),
+		end: todayEndOfDay,
+		displayKey: __( 'Year to date', 'woocommerce-payments' ),
+	},
+};
+
+const formatDateRange = (
+	start: moment.Moment,
+	end: moment.Moment
+): string => {
+	// Today - show only today's date.
+	if ( start.isSame( end, 'day' ) ) {
+		return start.format( 'MMMM D, YYYY' );
+	}
+
+	// Different years - show year for both start and end
+	if ( ! start.isSame( end, 'year' ) ) {
+		return `${ start.format( 'MMMM D, YYYY' ) } - ${ end.format(
+			'MMMM D, YYYY'
+		) }`;
+	}
+
+	// Same year - show year only for end date.
+	return `${ start.format( 'MMMM D' ) } - ${ end.format( 'MMMM D, YYYY' ) }`;
+};
+
+const options = Object.keys( timeOptions ).map( ( optionKey ) => {
+	const option = timeOptions[ optionKey ];
+	return {
+		key: optionKey,
+		name: option.displayKey,
+		hint: formatDateRange( option.start, option.end ),
+	};
+} );
+
+export const defaultDateRange = {
+	key: 'last_7_days',
+	date_start: timeOptions.last_7_days.start.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+	date_end: timeOptions.last_7_days.end.format( 'YYYY-MM-DD\\THH:mm:ss' ),
+};
+
+export const DateRangePicker: React.FC< {
+	onDateRangeChange: ( dateRange: DateRange ) => void;
+	selectedKey: string;
+} > = ( { onDateRangeChange, selectedKey } ) => {
+	return (
+		<InlineLabelSelect
+			label="Select an option"
+			options={ options }
+			value={ options.find( ( option ) => option.key === selectedKey ) }
+			placeholder="Select an option..."
+			onChange={ ( changes ) => {
+				const selectedItem = changes.selectedItem;
+				if ( selectedItem ) {
+					const start = timeOptions[ selectedItem.key ].start.format(
+						'YYYY-MM-DD\\THH:mm:ss'
+					);
+					const end = timeOptions[ selectedItem.key ].end.format(
+						'YYYY-MM-DD\\THH:mm:ss'
+					);
+					const { key } = selectedItem;
+					recordEvent( 'wcpay_overview_date_range_change', {
```

Comment:
This track name does not align with what's [registered](https://github.com/Automattic/tracks-events-registration/pull/2467/files#diff-9be6d918fbf3657afd4aa996cbd619f5dd632aa9f3dc728a92f55f1e611c4c27).

```suggestion
					recordEvent( 'wcpay_overview_payment_activity_period_change', {
```

Reply:
`wcpay_overview_payment_activity_period_*` is a clear context for the event – this "range change" applies only to WooPayments > Overview > Payment activity widget.

Reply:
>Looks good to me and tested well but I found discrepancy with tracking. Please check if my suggestion makes sense.

Definitely a big callout and a terrible oversight on my part. Thanks for catching 🙏

Reply:
The component and the recordEvent tracker have [moved ](https://github.com/Automattic/woocommerce-payments/blob/bd95962f5ceff8a1df703db832c0c33dff284199/client/components/payment-activity/index.tsx#L134)within the index.tsx file . I have fixed it here bd95962f5ceff8a1df703db832c0c33dff284199